YARCH/Sources/Modules/CatalogDetails/CatalogDetailsTableDataSource.swift

```diff
@@ -0,0 +1,32 @@
+import UIKit
+
+extension CatalogDetailsTableDataSource {
+    enum Configuration {
+        static let numberOfSections: Int = 1
+    }
+}
+
+class CatalogDetailsTableDataSource: NSObject, UITableViewDataSource {
+
+    var representableViewModels: [CatalogDetailsCellRepresentable]
+
+    init(viewModels: [CatalogDetailsCellRepresentable] = []) {
+        representableViewModels = viewModels
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return representableViewModels.count
+    }
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return Configuration.numberOfSections
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCellWithRegistration(type: CatalogDetailsCell.self, indexPath: indexPath)
+        guard let cellRepresentable = representableViewModels[safe: indexPath.row] else { return cell }
```

Comment:
Could be refactored to remove code duplication ("return cell"):
```
if let cellRepresentable = ... {
  cell.configure(...)
}
return cell
```

Reply:
Fixed!